csrc/sparse_attention/utils.cpp

```diff
@@ -94,14 +94,14 @@ ret_t sdd_segment(torch::Tensor layout, int start_width)
     // block index
     torch::Tensor idx = torch::zeros_like(layout);
     int current = 0;
-    size_t H = layout.size(0);
-    size_t M = layout.size(1);
-    size_t N = layout.size(2);
+    int64_t H = layout.size(0);
+    int64_t M = layout.size(1);
+    int64_t N = layout.size(2);
```

Comment:
Was this change required for CPU Adam+Adagrad support on windows?

Reply:
Oh I see, you're right here, this is fixing a bug I think. 

https://pytorch.org/cppdocs/api/classat_1_1_tensor.html#_CPPv4NK2at6Tensor4sizeEN2at7DimnameE

Reply:
I, honestly, forgot what the compile error was, but torch's C++ [size](https://pytorch.org/cppdocs/api/function_namespaceat_1a9dd1a76ff523b31c90a5b3c9c7a3dab8.html?highlight=size) explicitly returns `int64_t`.

Reply:
Can we use uint64_t instead, because if I remember correctly size_t is unsigned long int?